tests/linalg_test.py

```diff
@@ -277,14 +277,27 @@ def testEighGradVectorComplex(self, shape, dtype, rng, lower, eps):
     new_a = (new_a + onp.conj(new_a.T)) / 2
     # Assert rtol eigenvalue delta between perturbed eigenvectors vs new true eigenvalues.
     RTOL=1e-2
-    assert onp.max(
-      onp.abs((onp.diag(onp.dot(onp.conj((v+dv).T), onp.dot(new_a,(v+dv)))) - new_w) / new_w)) < RTOL
+    onp.testing.assert_allclose(
+        onp.diag(onp.dot(onp.conj((v+dv).T), onp.dot(new_a,(v+dv)))), new_w,
+        rtol=RTOL)
     # Redundant to above, but also assert rtol for eigenvector property with new true eigenvalues.
     assert onp.max(
       onp.linalg.norm(onp.abs(new_w*(v+dv) - onp.dot(new_a, (v+dv))), axis=0) /
       onp.linalg.norm(onp.abs(new_w*(v+dv)), axis=0)
     ) < RTOL
 
+  def testEighGradDegenerate(self):
+    rng = jtu.rand_default()
+    a = np.eye(2)
+    a_dot = a[:, ::-1]
+    (w, v), (dw, dv) = jvp(np.linalg.eigh, primals=(a,), tangents=(a_dot,))
+    # correct eigenvectors are 1/sqrt(2) * [[-1, 1], [1, 1]], up to arbitrary
+    # choice of phase
+    onp.testing.assert_allclose(abs(v), onp.ones((2, 2)) / onp.sqrt(2))
+    onp.testing.assert_allclose(w, onp.ones((2,)))
+    onp.testing.assert_allclose(abs(dv), onp.zeros((2, 2)))
+    onp.testing.assert_allclose(dw, onp.array([-1, 1]))
```

Comment:
TODO: more test cases. Currently nothing is checking eigenvector derivatives in the case of degeneracies if they are non-zero.